### PR TITLE
Import errors uniformization

### PR DESCRIPTION
**Pull request by yspringfield**
Fixed: #8452

### CODE REVIEW

File: libs/import/src/lib/pipes/is-disabled.pipe.ts [hunk 1]
```diff
@@ -4,8 +4,8 @@ import { errorCount } from './error-count.pipe';
 
 @Pipe({ name: 'isDisabled' })
 export class IsDisabledPipe implements PipeTransform {
-  transform(element: ImportState, processing: number) {
-    return errorCount(element) > 0 || processing > 0 || element.importing;
+  transform(element: ImportState, processing: number, imported=false) {
```
Comment (phpgeekfr) [inline libs/import/src/lib/pipes/is-disabled.pipe.ts:7]: bad formatting

File: libs/import/src/lib/table-extracted-elements/contracts/contracts.component.ts [hunk 2]
```diff
@@ -96,62 +89,20 @@ export class TableExtractedContractsComponent implements AfterViewInit {
...
-    const isOverlappingMandate = importState.contract.type === 'mandate' && availabilities.some(availability => availability.available.length);
-    return { isOverlappingSale, isOverlappingMandate };
-  }
-
   /**
    * Adds a contract to database and prevents multi-insert by refreshing mat-table
    * @param importState
    */
   private async add(importState: ContractsImportState, { increment } = { increment: false }) {
+    importState.imported=true;
```
Comment (phpgeekfr) [inline libs/import/src/lib/table-extracted-elements/contracts/contracts.component.ts:97]: bad formatting

File: libs/import/src/lib/table-extracted-elements/contracts/contracts.component.ts [hunk 3]
```diff
@@ -96,62 +89,20 @@ export class TableExtractedContractsComponent implements AfterViewInit {
...
+    return true;
```
Comment (phpgeekfr) [inline libs/import/src/lib/table-extracted-elements/contracts/contracts.component.ts:98]: ? 

the rest of the function will never be processed

File: libs/import/src/lib/utils.ts [hunk 4]
```diff
@@ -343,30 +344,23 @@ export function adminOnlyWarning<T = unknown>(value: T, name: string): ImportLog
...
   reason: string;
   field?: string;
   message: string;
   abstract type: 'warning' | 'error';
 
-  constructor(private value: T, options: LogOption) {
+  constructor(private value: T, options: SpreadsheetImportError) {
     super(options.message);
-    const { message, name, reason, field = '' } = options;
+    const { message='', name, reason, field = '' } = options;
```
Comment (phpgeekfr) [inline libs/import/src/lib/utils.ts:363]: bad formatting

File: libs/import/src/lib/utils.ts [hunk 5]
```diff
@@ -19,13 +19,14 @@ export interface SpreadsheetImportEvent {
...
   reason: string;
-  type: 'error' | 'warning';
+  type?: 'error' | 'warning';
   field?: string;
   message?: string;
 }
 
 export interface ImportState {
   errors?: SpreadsheetImportError[];
+  imported?:boolean;
```
Comment (phpgeekfr) [inline libs/import/src/lib/utils.ts:29]: bad formatting

File: libs/import/src/lib/view-extracted-elements/contract/utils.ts [hunk 6]
```diff
@@ -56,14 +58,47 @@ function toTerm(rawTerm: FieldsConfig['term'][number], contractId: string, fires
...
+async function getExistingContracts(type: 'sale' | 'mandate', titleId: string, contractService: ContractService, termService: TermService): Promise<(FullSale | FullMandate)[]> {
+  const query = getTitleContracts(type, titleId);
+  const contracts = await contractService.getValue(query);
+  const promises = contracts.map(contract => termService.getValue(contract.termIds))
+  const terms = await Promise.all(promises);
+  return contracts.map((contract, idx) => ({ ...contract, terms: terms[idx] }) as FullSale | FullMandate)
+}
+
+
+
```
Comment (phpgeekfr) [inline libs/import/src/lib/view-extracted-elements/contract/utils.ts:78]: extra lines not needed

File: libs/import/src/lib/view-extracted-elements/contract/utils.ts [hunk 7]
```diff
@@ -144,6 +179,24 @@ export async function formatContract(
       }
     }
 
+
+    const overlap = await verifyOverlappingMandatesAndSales(contract, terms, contractService, termService)
```
Comment (phpgeekfr) [inline libs/import/src/lib/view-extracted-elements/contract/utils.ts:183]:
```suggestion
    const overlap = await verifyOverlappingMandatesAndSales(contract, terms, contractService, termService);
```

File: libs/import/src/lib/view-extracted-elements/contract/utils.ts [hunk 8]
```diff
@@ -144,6 +179,24 @@ export async function formatContract(
...
+        reason: 'A term overlaps with that of an existing contract.',
+        message: 'A term overlaps with that of an existing contract.'
+      });
+    }
+    if (overlap.sale) {
+      errors.push({
+        type: 'error',
+        name: 'Contract',
+        reason: 'The terms of the imported sale have already sold been.',
+        message: 'The terms of the imported sale have already sold been.'
```
Comment (phpgeekfr) [inline libs/import/src/lib/view-extracted-elements/contract/utils.ts:197]: have already been sold ?

File: libs/import/src/lib/view-extracted-elements/contract/utils.ts [hunk 9]
```diff
@@ -144,6 +179,24 @@ export async function formatContract(
...
     }
 
+
+    const overlap = await verifyOverlappingMandatesAndSales(contract, terms, contractService, termService)
+    if (overlap.mandate) {
+      errors.push({
+        type: 'error',
+        name: 'Contract',
+        reason: 'A term overlaps with that of an existing contract.',
+        message: 'A term overlaps with that of an existing contract.'
```
Comment (phpgeekfr) [inline libs/import/src/lib/view-extracted-elements/contract/utils.ts:189]: not same sentence as previously, is it ok ?

Comment (yspringfield) [reply libs/import/src/lib/view-extracted-elements/contract/utils.ts:189]: Yes, had to shorten it as the previous sentence was getting cropped.

**Review by phpgeekfr — APPROVED** [review]
LGTM

**Pull request merged**